cmd/account/rotate-secret.go

```diff
@@ -80,128 +57,171 @@ func newRotateSecretOptions(streams genericclioptions.IOStreams, flags *genericc
 }
 
 func (o *rotateSecretOptions) complete(cmd *cobra.Command, args []string) error {
+
 	if len(args) != 1 {
-		return cmdutil.UsageErrorf(cmd, "IAM User name argument is required")
+		return cmdutil.UsageErrorf(cmd, "Account CR argument is required")
 	}
-	o.iamUsername = args[0]
 
-	// account CR name and account ID cannot be empty at the same time
-	if o.accountName == "" && o.accountID == "" {
-		return cmdutil.UsageErrorf(cmd, "AWS account CR name and AWS account ID cannot be empty at the same time")
-	}
+	o.accountCRName = args[0]
 
-	if o.accountName != "" && o.accountID != "" {
-		return cmdutil.UsageErrorf(cmd, "AWS account CR name and AWS account ID cannot be set at the same time")
+	if o.profile == "" {
+		o.profile = "default"
 	}
 
-	// only initialize kubernetes client when account name is set
-	if o.accountName != "" {
-		var err error
-		o.kubeCli, err = k8s.NewClient(o.flags)
-		if err != nil {
-			return err
-		}
+	var err error
+	o.kubeCli, err = k8s.NewClient(o.flags)
+	if err != nil {
+		return err
 	}
 
 	return nil
 }
 
 func (o *rotateSecretOptions) run() error {
+
 	ctx := context.TODO()
 	var err error
-	awsSetupClient, err := awsprovider.NewAwsClient(o.profile, o.region, o.cfgFile)
+
+	// Get the associated Account CR from the provided name
+	var accountID string
+	account, err := k8s.GetAWSAccount(ctx, o.kubeCli, common.AWSAccountNamespace, o.accountCRName)
 	if err != nil {
 		return err
 	}
 
-	var accountID string
-	if o.accountName != "" {
-		account, err := k8s.GetAWSAccount(ctx, o.kubeCli, o.accountNamespace, o.accountName)
-		if err != nil {
-			return err
-		}
-		accountID = account.Spec.AwsAccountID
-	} else {
-		accountID = o.accountID
-	}
+	// Set the account ID
+	accountID = account.Spec.AwsAccountID
 
-	callerIdentityOutput, err := awsSetupClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	// Use provided profile
+	awsSetupClient, err := awsprovider.NewAwsClient(o.profile, "", "")
 	if err != nil {
 		return err
 	}
 
-	roleArn := aws.String(fmt.Sprintf("arn:aws:iam::%s:role/%s", accountID, awsv1alpha1.AccountOperatorIAMRole))
-	credentials, err := awsprovider.GetAssumeRoleCredentials(awsSetupClient, aws.Int64(900),
-		callerIdentityOutput.UserId, roleArn)
+	// Ensure AWS calls are succesful with client
+	callerIdentityOutput, err := awsSetupClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
```

Comment:
guess the best way to check if aws calls are successful is to make an actual aws call :) this works, seems.. hacky though

Reply:
This call was already in this command. Why do you think its hacky?

Reply:
IMO get-caller-identity is a valid way of checking if the credentials are good.